connectors/tests/test_job_cleanup.py

```diff
@@ -0,0 +1,82 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+
+import asyncio
+from unittest.mock import AsyncMock, Mock, patch
+
+import pytest
+
+from connectors.services.job_cleanup import JobCleanUpService
+from connectors.tests.test_byoc import AsyncIterator
+
+CONFIG = {
+    "elasticsearch": {
+        "host": "http://nowhere.com:9200",
+        "user": "elastic",
+        "password": "changeme",
+    },
+    "service": {
+        "max_errors": 20,
+        "max_errors_span": 600,
+        "job_cleanup_interval": 1,
+    },
+    "native_service_types": ["mongodb"],
+}
+
+
+def create_service():
+    return JobCleanUpService(CONFIG)
+
+
+def mock_connector():
+    connector = Mock()
+    connector.id.return_value = "1"
+    connector._sync_done = AsyncMock()
+    return connector
+
+
+def mock_sync_job():
+    job = Mock()
+    job.job_id.return_value = "1"
+    job.index_name.return_value = "index_name"
+    job.connector_id.return_value = "1"
+    return job
+
+
+@pytest.mark.asyncio
+@patch("connectors.byoc.SyncJobIndex.delete_jobs")
+@patch("connectors.byoc.SyncJobIndex.delete_indices")
+@patch("connectors.byoc.SyncJobIndex.stuck_jobs")
+@patch("connectors.byoc.SyncJobIndex.orphaned_jobs")
+@patch("connectors.byoc.ConnectorIndex.fetch_by_id")
+@patch("connectors.byoc.ConnectorIndex.supported_connectors")
+@patch("connectors.byoc.ConnectorIndex.all_connectors")
+async def test_cleanup_jobs(
+    all_connectors,
+    supported_connectors,
+    fetch_by_id,
+    orphaned_jobs,
+    stuck_jobs,
+    delete_indices,
+    delete_jobs,
+):
+    connector = mock_connector()
+    sync_job = mock_sync_job()
+
+    all_connectors.return_value = AsyncIterator([connector])
+    supported_connectors.return_value = AsyncIterator([connector])
+    fetch_by_id.return_value = connector
+    orphaned_jobs.return_value = AsyncIterator([sync_job])
+    stuck_jobs.return_value = AsyncIterator([sync_job])
+    delete_jobs.return_value = {"deleted": 1, "failures": [], "total": 1}
+
+    service = create_service()
+    asyncio.get_event_loop().call_later(0.5, service.stop)
+    await service.run()
+
+    assert delete_indices.called
```

Comment:
Do you think it makes sense to also check that the arguments are correct using `call_args_list`?

Reply:
Done in https://github.com/elastic/connectors-python/pull/219/commits/9efd2e06620b596ff592e1c81994fdbf302ed348